### PR TITLE
[10] product quick bom

### PR DESCRIPTION
**Pull request by EBII**
This module was written to create quickly the bom for your product.
Indeed in the product form you now have a new tab "bom" that give you the posibility
to add the bom line directly here

### CODE REVIEW

File: product_quick_bom/README.rst [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+=================
+
+This module was written to create quickly the bom for your product.
+Indeed in the product form you now have a new tab "bom" that give you the posibility
+to add the bom line directly here
+
+Installation
+============
+
+To install this module, you just need to install it
```
Comment (LoisRForgeFlow) [inline product_quick_bom/README.rst]: please remove not needed sections.

File: product_quick_bom/README.rst [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+Configuration
+=============
+
+No configuration is required
```
Comment (LoisRForgeFlow) [inline product_quick_bom/README.rst]: Also this one.

File: product_quick_bom/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        "python": [],
+        "bin": [],
+    },
+    "depends": [
+        "mrp",
+    ],
+    "data": [
+        "product_view.xml",
+    ],
+    "demo": [
```
Comment (LoisRForgeFlow) [inline product_quick_bom/__manifest__.py]: remove empty keys.

File: product_quick_bom/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    "name": "Product Quick Bom",
+    "summary": "Create the bom directly from the product",
+    "version": "10.0.1.2.0",
+    "category": "mrp",
+    "website": "https://odoo-community.org/",
+    "author": "Akretion, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "application": False,
+    "installable": True,
+    "external_dependencies": {
```
Comment (LoisRForgeFlow) [inline product_quick_bom/__manifest__.py]: remove empty keys

File: product_quick_bom/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+#   Copyright (C) 2015 Akretion (http://www.akretion.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Product Quick Bom",
+    "summary": "Create the bom directly from the product",
+    "version": "10.0.1.2.0",
```
Comment (LoisRForgeFlow) [inline product_quick_bom/__manifest__.py]: Isn't this the first version to be merged in v10? Please update to 10.0.1.0.0.

**Comment by bealdav** [comment]
please remove .pot file as defined in guideline

File: product_quick_bom/bom.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+from odoo import models, fields
+
+
+class MrpBom(models.Model):
+    _inherit = 'mrp.bom'
+
+    _sql_constraint = (
+        'uniq_product_template',
+        'uniq(product_tmpl_id)',
+        'You can only have one Bom per product template',
```
Comment (Garamotte) [inline product_quick_bom/bom.py]: Add `_()` around the error message to make it translatable.

File: product_quick_bom/bom.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        )
+
+
+class MrpBomLine(models.Model):
+    _inherit = 'mrp.bom.line'
+
+    product_tmpl_id = fields.Many2one(
+        'product.template',
+        related='bom_id.product_tmpl_id',
+        store=True)
```
Comment (Garamotte) [inline product_quick_bom/bom.py]: I think you forgot to set this field `readonly=True`.
Also, please explicitely name all parameters (`comodel_name='product.template'`).

File: product_quick_bom/product.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+from odoo import models, fields, api
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    bom_line_ids = fields.One2many(
+        'mrp.bom.line',
+        'product_tmpl_id',
```
Comment (Garamotte) [inline product_quick_bom/product.py]: Explicitely name all parameters.

File: product_quick_bom/product.py [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+             'here the componant to product them')
+
+    @api.model
+    def _extract_bom_line(self, vals):
+        return vals.pop('bom_line_ids', {})
+
+    @api.multi
+    def _prepare_bom_vals(self, vals):
+        return {
+            'product_tmpl_id': self.id,
```
Comment (Garamotte) [inline product_quick_bom/product.py]: Add `self.ensure_one()` as first line of the method to ensure you have only one record in self.

File: product_quick_bom/product_view.xml [hunk 10]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+<data>
```
Comment (Garamotte) [inline product_quick_bom/product_view.xml]: Remove `<data>` tag.

File: product_quick_bom/product_view.xml [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+        <record id="product_template_form_view" model="ir.ui.view">
+            <field name="model">product.template</field>
+            <field name="inherit_id" ref="product.product_template_form_view"/>
+            <field name="priority" eval="200"/>
```
Comment (Garamotte) [inline product_quick_bom/product_view.xml]: Defining the priority is really needed here ?

File: product_quick_bom/README.rst [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+=====
+
+To use this module, you need to:
+
+* go to a product form, click on edit, add a line in the bom tab and see that a bom
+  linked to the product have been created
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/{repo_id}/{branch}
```
Comment (Garamotte) [inline product_quick_bom/README.rst]: Replace `{repo_id}` by the real repo id and remove the comment.

File: product_quick_bom/bom.py [hunk 13]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+#   Copyright (C) 2015 Akretion (http://www.akretion.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields
```
Comment (Garamotte) [inline product_quick_bom/bom.py]: Imports should be alphabetized.

File: product_quick_bom/product.py [hunk 14]
```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+#   Copyright (C) 2015 Akretion (http://www.akretion.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields, api
```
Comment (Garamotte) [inline product_quick_bom/product.py]: Imports should be alphabetized.

File: product_quick_bom/bom.py [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+    _inherit = 'mrp.bom'
+
+    _sql_constraint = (
+        'uniq_product_template',
+        'uniq(product_tmpl_id)',
+        'You can only have one Bom per product template',
+        )
+
+
+class MrpBomLine(models.Model):
```
Comment (Garamotte) [inline product_quick_bom/bom.py]: Split one file per model.

**Review by Garamotte — CHANGES_REQUESTED** [review]

File: product_quick_bom/product_view.xml (hunk 11 above)
Comment (EBII) [reply product_quick_bom/product_view.xml]: i don't know

File: product_quick_bom/README.rst (hunk 12 above)
Comment (EBII) [reply product_quick_bom/README.rst]: thanks

File: product_quick_bom/product.py [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    bom_line_ids = fields.One2many(
+        comodel_name='mrp.bom.line',
+        inverse_name='product_tmpl_id',
+        string='Bom Line',
+        help='If you product is manufactured you can select'
+             'here the componant to product them')
```
Comment (bguillot) [inline product_quick_bom/product.py:16]: s/you/your
add a space after "select"
s/componant/component
s/product/produce
s/them/it

And change the po file accordingly

Thanks !

Comment (EBII) [reply product_quick_bom/product.py:16]: replace by https://github.com/OCA/manufacture/pull/183

**Pull request closed without merging**